nautobot/core/testing/filters.py

```diff
@@ -286,6 +292,20 @@ def test_q_filter_valid(self):
                         obj, obj_field_name = self._get_nested_related_obj_and_its_field_name(obj, obj_field_name)
                     self._assert_q_filter_predicate_validity(obj, obj_field_name, filter_field_name, lookup_method)
 
+        def test_content_type_related_fields_uses_content_type_filter(self):
+            for field in self.queryset.model._meta.fields:
+                related_model = getattr(field, "related_model", None)
+                if not related_model or related_model != ContentType:
+                    continue
+                with self.subTest(
+                    f"Assert {self.filterset.__class__.__name__}.{field.name} implements ContentTypeFilter"
+                ):
+                    filter_field = self.filterset.get_filters().get(field.name)
+                    if not filter_field:
+                        # This field is not part of the Filterset.
+                        continue
+                    self.assertIsInstance(filter_field, (ContentTypeFilter, ContentTypeMultipleChoiceFilter))
```

Comment:
Looking at the code maybe we need to permit `ContentTypeChoiceFilter` as well?

I gotta wonder why we have all of `ContentTypeFilter`, `ContentTypeChoiceFilter`, and `ContentTypeMultipleChoiceFilter` - naively I would expect `ContentTypeMultipleChoiceFilter` in all cases. Guess that'd be a breaking change to implement at this time though.